_posts/2018-07-27-Our-first-look-at-debos-new-Debian-images-generator.md

```diff
@@ -0,0 +1,285 @@
+
+```
+./docker/run.sh -t architecture:"amd64" doc/examples/example.yaml
+```
+
+```
+2018/07/26 18:36:39 Debootstrap (stage 2) | chroot: failed to run command '/debootstrap/debootstrap': Exec format error
+2018/07/26 18:36:39 debootstrap.log | gpgv: Signature made Thu Jul 26 14:21:51 2018 UTC
+2018/07/26 18:36:39 debootstrap.log | gpgv:                using RSA key A1BD8E9D78F7FE5C3E65D8AF8B48AD6246925553
+2018/07/26 18:36:39 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (7.0/wheezy) <ftpmaster@debian.org>"
+2018/07/26 18:36:39 debootstrap.log | gpgv: Signature made Thu Jul 26 14:21:51 2018 UTC
+2018/07/26 18:36:39 debootstrap.log | gpgv:                using RSA key 126C0D24BD8A2942CC7DF8AC7638D0442B90D010
+2018/07/26 18:36:39 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (8/jessie) <ftpmaster@debian.org>"
+2018/07/26 18:36:39 debootstrap.log | gpgv: Signature made Thu Jul 26 14:21:51 2018 UTC
+2018/07/26 18:36:39 debootstrap.log | gpgv:                using RSA key 16E90B3FDF65EDE3AA7F323C04EE7237B7D453EC
+2018/07/26 18:36:39 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org>"
+2018/07/26 18:36:39 Action `debootstrap` failed at stage Run, error: exit status 126
+Powering off.
+```
+
+And the build for host architecture:
+
+```
+./docker/run.sh -t architecture:"amd64" doc/examples/example.yaml
+```
+
+Finished just fine:
+
+```
+2018/07/26 19:20:30 ==== overlay ====
+Overlaying /root/doc/examples/overlays/sudo on /scratch/root
+2018/07/26 19:20:30 ==== run ====
+2018/07/26 19:20:30 echo debian > /etc/hostname | host's /etc/localtime is not a symlink, not updating container timezone.
+2018/07/26 19:20:30 ==== pack ====
+2018/07/26 19:20:30 Compression to /root/debian-stretch-amd64.tgz
+Powering off.
+2018/07/26 19:20:44 ==== Recipe done ====
+```
+
+So, when building for non-host architecture, we have the same error as was in
+case of `VirtualBox`. I'm not sure what might be the issue: either
+the proper `qemu-static` is not used, or some virtualization problems? Debugging
+of what's going on beneath is not easy, especially if there is no way to have
+more debug output from the `debos` or `fakemachine` tools (at least I have not
+found any).
+
+## Conclusion
+
+`debos` seems like a really cool tool for Debian images building without
+tinkering with `debootrap` and `chroot` script that much. Unfortunately, I was
+unable to build an image for `arm` or `arm64` so far (which is my main
+interest). Maybe on the native installed `Debian` it would just work perfectly
+I did not have the opportunity to try it this way. I think that having a docker
+container with `debos` tool fits perfectly into this case and would allow many
+more people to benefit from using it. I will try to push my work upstream and
+discuss, so the cross-building issues will be hopefully resolved.
```

Comment:
`... and discuss...`
Subject missing. You can say, for example: `... and discuss this issue...` or `... and start a discussion`. Moreover, you can invite readers to this discussion once you start it.